spec/src/language_negotiation_lambda_spec.js

```diff
@@ -34,9 +31,62 @@ describe( 'languageNegotiationLambda', function() {
     var default_header = ['en'];
 
     beforeEach(function(){
+        delete event.Records[0].cf.request.headers["Cookie"];
         delete event.Records[0].cf.request.headers["Accept-Language"];
         delete event.Records[0].cf.request.headers["X-Accept-Language"];
     });
+
+    describe( 'Well formed cookies', function(){
+        [
+            { cookie: "en",     expected: "en" },
+            { cookie: "EN",     expected: "en" },
+            { cookie: "es",     expected: "es" },
+            { cookie: "ES",     expected: "es" },
+            { cookie: "en-US",  expected: "en" },
+            { cookie: "es-419", expected: "es" }
+        ].forEach( function( cookies ) {
+            it( `Resolves cookie ${cookies.cookie} to ${cookies.expected}`, function( done ) {
+                event.Records[0].cf.request.headers.Cookie = [
+                    process.env.LANGUAGE_COOKIE_NAME+"="+cookies.cookie
+                ];
+
+                languageNegotiationLambda.handler( event, { /* context */ }, (err, result) => {
+                    try {
+                        expect( err ).toBeNull();
+                        expect( result.headers['X-Accept-Language'] ).toEqual([cookies.expected]);
+                        done();
+                    }
+                    catch( error ) {
+                        done( error );
+                    }
+                });
+            });
+        });
+
+        [
+            "fr",
+            ""
+        ].forEach( function( cookie ) {
```

Comment:
Same comment in https://github.com/edx/language-negotiation-lambda/pull/8. I think these tests can be combined with the ones above it since it runs through the same code and has a valid output.